### PR TITLE
Added regex search

### LINKED ISSUE #5786: Regex/Boolean search in filter fields
**Regex/Boolean search in filterfields**  

Thanks for the wonderful framework!! I explored bootstrap-table and been using this awesome framework for quite a while.

It would be nice if regex search for filter field is implemented. Use case as below

- column in table contains strings 1asdf45_0099 and 25asd45_1888
- search on filter should return all strings with characters '45' (both of the above strings)

I can see some regex parsing/search in the code of bootstrap-tables but not for this use case.

### PR DESCRIPTION
**Pull request by UtechtDustin**
**🤔Type of Request**
- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Improvement**
- [ ] **Documentation**
- [ ] **Other**

**🔗Resolves an issue?**

Fix #5786 

**📝Changelog**

- [x] **Core**
- [x] **Extensions**

Added a new search option which allows to filter the table using regex.

**💡Example(s)?**

Example: https://live.bootstrap-table.com/code/UtechtDustin/8414
e.g. search for `[57]$` to get the rows which data ends with `5`  or `7`.

**☑️Self Check before Merge**

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed (https://github.com/wenzhixin/bootstrap-table-examples/pull/439)
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: site/docs/api/table-options.md [hunk 1]
```diff
@@ -1074,6 +1074,23 @@ The table options are defined in `jQuery.fn.bootstrapTable.defaults`.
...
+- **Detail:**
+
+  Set `true` to enable the regex search.
+  Is this option enabled, you can search with regex e.g. `[47a]$` for all items which ends with a `4`, `7` or `a`.     
+  The regex can be entered with `/[47a]$/gim` or without `[47a]$` flags, the default flags are `gim`.
+
+
+- **Default:** `false`
+
+- **Example:** [Remember Order](https://examples.bootstrap-table.com/#options/regex-search.html)
```
Comment (djhvscf) [inline site/docs/api/table-options.md:1092]: I think it is: **Regex Search** instead of **Remember Order**?

**Comment by UtechtDustin** [comment]
The linter throw the following errors:
```
src/themes/_theme.scss
 1:1  ✖  Unknown rule declaration-block-no-duplicate-custom-properties  declaration-block-no-duplicate-custom-properties
 1:1  ✖  Unknown rule named-grid-areas-no-invalid                       named-grid-areas-no-invalid                     
 1:1  ✖  Unknown rule no-invalid-position-at-import-rule                no-invalid-position-at-import-rule              
 1:1  ✖  Unknown rule no-irregular-whitespace                           no-irregular-whitespace
``` 

But i don't event modified the file, that error also occurs for  #5528.
Any idea what's wrong ?

File: src/utils/index.js [hunk 2]
```diff
@@ -193,6 +193,25 @@ export default {
     return true
   },
 
+  regexCompare (value, search) {
+    try {
+      const regexpParts = search.match(/^\/(.*?)\/([gim]*)$/)
+      let regex
+
+      if (regexpParts) {
```
Comment (djhvscf) [inline src/utils/index.js:201]: What about to use the ternary condition?

File: src/utils/index.js [hunk 3]
```diff
@@ -193,6 +193,25 @@ export default {
...
+        regex = new RegExp(regexpParts[1], regexpParts[2])
+      } else {
+        regex = new RegExp(search, 'gim')
+      }
+
+      if (value.toString().search(regex) !== -1) {
+        return true
+      }
+    } catch (e) {
```
Comment (djhvscf) [inline src/utils/index.js:210]: Which kind of exceptions can we have here? Is it worth to throw it and display it in the dev console?

File: src/utils/index.js [hunk 4]
```diff
@@ -193,6 +193,25 @@ export default {
...
+      const regexpParts = search.match(/^\/(.*?)\/([gim]*)$/)
+      let regex
+
+      if (regexpParts) {
+        regex = new RegExp(regexpParts[1], regexpParts[2])
+      } else {
+        regex = new RegExp(search, 'gim')
+      }
+
+      if (value.toString().search(regex) !== -1) {
```
Comment (djhvscf) [inline src/utils/index.js:207]: Can be 
```js
if (value.toString().search(regex) > -1) {
```

**Review by djhvscf — CHANGES_REQUESTED** [review]

File: site/docs/api/table-options.md (hunk 1 above)
Comment (UtechtDustin) [reply site/docs/api/table-options.md:1092]: done

File: src/utils/index.js (hunk 2 above)
Comment (UtechtDustin) [reply src/utils/index.js:201]: done

File: src/utils/index.js (hunk 4 above)
Comment (UtechtDustin) [reply src/utils/index.js:207]: `!==` is more strict, so it's ok i guess.
But if you wish i also can change it to `>`.

**Comment by djhvscf** [comment]
> The linter throw the following errors:
> 
> ```
> src/themes/_theme.scss
>  1:1  ✖  Unknown rule declaration-block-no-duplicate-custom-properties  declaration-block-no-duplicate-custom-properties
>  1:1  ✖  Unknown rule named-grid-areas-no-invalid                       named-grid-areas-no-invalid                     
>  1:1  ✖  Unknown rule no-invalid-position-at-import-rule                no-invalid-position-at-import-rule              
>  1:1  ✖  Unknown rule no-irregular-whitespace                           no-irregular-whitespace
> ```
> 
> But i don't event modified the file, that error also occurs for #5528.
> Any idea what's wrong ?

No idea. that .scss file is problematic. @wenzhixin any idea?

File: src/utils/index.js (hunk 4 above)
Comment (djhvscf) [reply src/utils/index.js:207]: Let's leave it as !==.. It's just the way we code

File: src/utils/index.js (hunk 3 above)
Comment (UtechtDustin) [reply src/utils/index.js:210]: As example `[a` is no valid regex as the square bracket must be closed (`[a]`).
So it is possible that we run in the catch if the user don't write quickly, so its fine that we dont display it in the dev console.
Also its no critical error i guess.

File: src/utils/index.js (hunk 4 above)
Comment (UtechtDustin) [reply src/utils/index.js]: We often use that in the bootstrap-table as example here: https://github.com/wenzhixin/bootstrap-table/blob/develop/src/bootstrap-table.js#L448
So lets leave it as `!==`.

**Comment by UtechtDustin** [comment]
> No idea. that .scss file is problematic. @wenzhixin any idea?
Found the problem, the stylelint package was not up2date.
After a update to `^13.13.1` its working fine now :-)

**Review by djhvscf — APPROVED** [review]

**Pull request merged**